snapcraft/parts/lifecycle.py

```diff
@@ -129,24 +138,29 @@ def run(command_name: str, parsed_args: "argparse.Namespace") -> None:
     emit.trace(f"command: {command_name}, arguments: {parsed_args}")
 
     snap_project = get_snap_project()
-    yaml_data = process_yaml(snap_project.project_file)
-    parse_info = _extract_parse_info(yaml_data)
+    yaml_data = load_yaml(snap_project.project_file)
+    build_plan = get_build_plan(yaml_data)
 
-    if parsed_args.provider:
-        raise errors.SnapcraftError("Option --provider is not supported.")
+    for build_on, build_to in build_plan:
```

Comment:
this is missing a check for managed mode

Reply:
What should be done differently for managed mode?

Reply:
I thought we were doing one container per build-to; is that not how charmcraft does it?

Reply:
Ah, finally I understand this.  Now, I'm creating one container for each `build-to`.  
The logic in snapcraft is a bit simpler than charmcraft, because charmcraft also checks against `base.name` and `base.channel`.